ArduCopter/defines.h

```diff
@@ -106,6 +106,20 @@ enum autopilot_modes {
     BRAKE =        17   // full-brake using inertial/GPS system, no pilot input
 };
 
+enum mode_reason_t {
+    MODE_REASON_UNKNOWN=0,
```

Comment:
Obviously not my call but I'll leave my opinion. I don't like this reason unknown:

- globally it helps developers being lazy by using unknown for any reason that isn't defined already
- its use now seems to be limited to changes from flight control, I would remove this unknown and add FLIGHT_CONTROL or something like that

You are using 0 for log in Plane/Rover and I would leave it like that, just don't document it so people don't use it